libs/user-identity-service/src/test/java/uk/gov/di/ipv/core/library/service/UserIdentityServiceTest.java

```diff
@@ -1088,6 +1089,30 @@ void generateUserIdentityShouldNotSetExitCodeWhenBreachingCiThresholdAndExitCode
         assertNull(userIdentity.getExitCode());
     }
 
+    @Test
+    void generateUserIdentityShouldSetRequiredExitCodeWhenP0AndNotBreachingCiThreshold()
+            throws Exception {
+        when(mockConfigService.getSsmParameter(CORE_VTM_CLAIM)).thenReturn("mock-vtm-claim");
+        when(mockConfigService.enabled(EXIT_CODES)).thenReturn(true);
+        when(mockConfigService.getSsmParameter(CI_SCORING_THRESHOLD)).thenReturn("10");
+        when(mockConfigService.getSsmParameter(EXIT_CODES_NON_CI_BREACHING_P0)).thenReturn("🐧");
+
+        when(mockConfigService.getContraIndicatorConfigMap())
+                .thenReturn(Map.of("X01", new ContraIndicatorConfig("X01", 4, -3, "1")));
+
+        ContraIndicators contraIndicators =
+                ContraIndicators.builder()
+                        .contraIndicatorsMap(
+                                Map.of("X01", ContraIndicator.builder().code("X01").build()))
+                        .build();
+
+        UserIdentity userIdentity =
+                userIdentityService.generateUserIdentity(
+                        USER_ID_1, "test-sub", "P0", contraIndicators);
+
+        assertEquals(List.of("🐧"), userIdentity.getExitCode());
+    }
```

Comment:
Do you think it'd be useful to make explicit that it _doesn't_ include an 'always included' code either?

Reply:
I supposed I could assert that there's no call to the config service to retrieve the "always included" code. I can't assert it's not directly in the returned list without defining what that code actually is (in this test case). The only way to do that is to mock the response for that call to the config service. But then if we don't actually call it (which we don't), then the tests get angry about unnecessary stubbings 🙂.

I'll add the negative assertion I think...

Reply:
Ah, yeah that's a bit annoying, but makes sense, thanks.